### DISCUSSION TITLE
Flutter & Dart - Faire du TDD avec Mockito

### DISCUSSION THREAD

**Discussion by giscus** [zero-filtre/zerofiltre-blog#174] [Announcements]
# Flutter & Dart - Faire du TDD avec Mockito

Apprenez à écrire du code propre et maintenable avec le TDD en utilisant Mockito dans vos projets Flutter.

https://zerofiltre.tech/articles/253

**Comment by imphilippesimo** [comment]
Je me rends compte que pas besoin de mockito dans cet article.

Tu as une classe à terter (*Class under test*) = `Password` *qu'il ne faut surtout pas mocker*, sinon tu serais en train de tester un mock. 
Ce qui n'est pas le but.

On mock une dépendance de la class under test. 

Ex: Si `Password` dépendait d'un élément `Crypto` pour générer des random par exemple, on aurait mocké `Crypto`.


Par conséquent, j'ai corrigé le code présent dans cet article et je te propose de faire les modifications suivantes afin qu'il soit validé:

- Ne plus mentionner tout ce qui a trait à mockito, pas besoin dans cet articles (librairies, génération ...)
- Corriger ton code source et en faire un pull request comme d'hab
- Migrer l'objectif de ton article vers : *Dans cet article, nous allons comprendre ce qu’est le TDD et comment l’implémenter dans un exemple avec le langage Dart* ~grâce au package Mockito~.

**Reply by imphilippesimo** [reply]
J'ai failli oublier : je te suggère de faire une image d'article plus belle et ceci très facilement grâce à [Canva](https://www.canva.com/)
